src/common/controllers/ads-controller.js

```diff
@@ -339,6 +367,39 @@ class AdsController extends FakeEventTarget implements IAdsController {
     });
   }
 
+  _handleReturnToLive(adBreaks: Array<RunTimeAdBreakObject>) {
+    this._liveEventManager.listenOnce(this._player, AdEventType.AD_ERROR, () => {
+      this._pushNextAdsForLive(adBreaks);
+      this._liveEventManager.removeAll();
+    });
+    this._liveEventManager.listenOnce(this._player, AdEventType.AD_BREAK_END, () => {
+      this._liveEventManager.listenOnce(this._player, Html5EventType.SEEKED, () => {
+        this._pushNextAdsForLive(adBreaks);
```

Comment:
why we have to wait for ad break done before pushing the next?

Reply:
Because you will seek to the live edge, and you don't know when to position the next ad before it.